src/py42/services/cases.py

```diff
@@ -0,0 +1,207 @@
+            that each contain a page of cases.
+        """
+        return get_all_pages(
+            self.get_page,
+            u"cases",
+            name=name,
+            status=status,
+            created_at=created_at,
+            updated_at=updated_at,
+            assignee=assignee,
+            subject=subject,
+            page_number=page_number,
+            page_size=page_size,
+            sort_direction=sort_direction,
+            sort_key=sort_key,
+            **kwargs
+        )
+
+    def get_case(self, case_number):
+        """Retrieve case details by case number.
+
+        Args:
+            case_number (int): Case number of the case.
+
+        Returns:
+            :class:`py42.response.Py42Response`
+        """
+        return self._connection.get("{}/{}".format(self._uri_prefix, case_number))
+
+    def export_summary(self, case_number):
+        """Download case summary in a PDF file.
+
+        Args:
+            case_number (int): Case number of the case.
+
+        Returns:
+            :class:`py42.response.Py42Response`
+        """
+        uri_prefix = u"{}/{}/{}".format(self._uri_prefix, case_number, u"export")
+        return self._connection.get(uri_prefix)
+
+    def update(
+        self,
+        case_number,
+        name=None,
+        subject=None,
+        assignee=None,
+        description=None,
+        findings=None,
+        status=None,
+    ):
+        """Update case details for the given case number.
+
+        Args:
+            case_number (int): Case number of the case.
+            name (str, optional): Name of the case. Defaults to empty string.
+            subject (str, optional): A subject of the case. Defaults to empty string.
+            assignee (str, optional): User UID of the assignee. Defaults to empty string.
+            description (str, optional): Description of the case. Defaults to empty string.
+            findings (str, optional): Notes on the case. Defaults to empty string.
+            status (str, optional): Status of the case. `OPEN` or `CLOSED`. Defaults to None.
+        Returns:
+            :class:`py42.response.Py42Response`
+        """
+
+        current_case_data = self.get_case(case_number).data
+
+        data = {
+            u"assignee": assignee or current_case_data.get(u"assignee"),
+            u"description": description or current_case_data.get(u"description"),
+            u"findings": findings or current_case_data.get(u"findings"),
+            u"name": name or current_case_data.get(u"name"),
+            u"subject": subject or current_case_data.get(u"subject"),
+            u"status": status or current_case_data.get("status"),
+        }
+        try:
+            return self._connection.put(
+                u"{}/{}".format(self._uri_prefix, case_number), json=data
+            )
+        except Py42BadRequestError as err:
+            if u"NO_EDITS_ONCE_CLOSED" in err.response.text:
+                raise Py42InvalidActionError(
```

Comment:
See https://github.com/code42/py42/pull/264/files#r553466324

This is what should be:

```
if u"NO_EDITS_ONCE_CLOSED" in err.response.text:
  raise Py42UpdateClosedCaseError()
raise
```

Reply:
Changed.